source/core_sntp_serializer.c

```diff
@@ -259,50 +292,96 @@ static SntpStatus_t calculateClockOffset( const SntpTimestamp_t * pClientTxTime,
     SntpStatus_t status = SntpSuccess;
 
     /* Variable for storing the first-order difference between timestamps. */
-    uint32_t firstOrderDiff = 0;
+    uint32_t firstOrderDiffSend = 0;
+    uint32_t firstOrderDiffRecv = 0;
+    bool sendPolarity, recvPolarity;
 
     assert( pClientTxTime != NULL );
     assert( pServerRxTime != NULL );
     assert( pServerTxTime != NULL );
     assert( pClientRxTime != NULL );
     assert( pClockOffset != NULL );
 
-    /* Calculate a sample first order difference value between the
-     * server and system timestamps. */
-    firstOrderDiff = pClientRxTime->seconds - pServerTxTime->seconds;
+    /* On-wire protocol formula's polarity for first order difference in the send network
```

Comment:
Comment mention about sendPolarity but the formula is calculating recvPolarity.

Reply:
Fixed